### PR TITLE
Json view schema

### PR DESCRIPTION
**Pull request by radovanradic**
Tried to work on creating JSON VIEW during schema creation. Faced new issues that were not expected. Namely, in order to create JSON VIEW, all table constraints and primary keys must be created. However, our table creation process don't include constraint creation (for some reason) and for that we cannot create JSON VIEW during schema creation.
Workaround for this test is to run migration via flyway, which is probably not what we want. Running this just to see if build will pass since was able to use new oracle container with 23c db.

### CODE REVIEW

File: data-jdbc/src/test/groovy/io/micronaut/data/jdbc/oraclexe/jsonview/entities/StudentScheduleView.java [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
 @Serdeable
+@JsonData(table = "TBL_STUDENT_CLASSES", permissions = "INSERT UPDATE DELETE")
 public class StudentScheduleView {
     @Id
     private Long id;
 
     @JsonProperty("class")
     @Relation(Relation.Kind.ONE_TO_ONE)
+    @JoinColumn(referencedColumnName = "CLASS_ID", name = "ID")
+    @JsonViewColumn(name = "class")
```
Comment (dstepanov) [inline data-jdbc/src/test/groovy/io/micronaut/data/jdbc/oraclexe/jsonview/entities/StudentScheduleView.java:21]: Why not to use `@MappedProperty` to define the column name?

**Comment by dstepanov** [comment]
Maybe we can support running the view creation using the flyway java migrations https://flywaydb.org/documentation/tutorials/java

File: data-jdbc/src/test/groovy/io/micronaut/data/jdbc/oraclexe/jsonview/entities/StudentScheduleView.java (hunk 1 above)
Comment (radovanradic) [reply data-jdbc/src/test/groovy/io/micronaut/data/jdbc/oraclexe/jsonview/entities/StudentScheduleView.java:21]: We could use that, but needed to introduce `@JsonColumnView` to mark if column has attributes/permissions `INSERT UPDATE DELETE CHECK` so thought to use it for this as well.

**Comment by radovanradic** [comment]
> Maybe we can support running the view creation using the flyway java migrations https://flywaydb.org/documentation/tutorials/java

This doesn't differ much from our schema creation, basically we would need to generate SQL for execution and don't think this gives us benefit. The main issue remains, we don't have foreign keys created and views cannot be created then. I have created new [PR to enable json view tests](https://github.com/micronaut-projects/micronaut-data/pull/2214) and will close this one. Don't think we should change schema creation stuff now when 4.0 would be going out and it can take a while or introduce new issues. For now we have json view support and tests, users can use flyway to create schema.

**Pull request closed without merging**